File: Domain/Account.cs
using Domain.Enums;

namespace Domain
{
    public class Account
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public decimal Balance { get; set; }
        public int CurrencyId { get; set; }
        public DateTime CreatedAt { get; set; }
        public string Description { get; set; }
        public AccountStatus Status { get; set; } = AccountStatus.Visible;
        public string UserId { get; set; }

        public virtual Currency Currency { get; set; }
        public virtual User User { get; set; }
        public virtual ICollection<Transaction> Transactions { get; set; }
        public virtual ICollection<Loan> Loans { get; set; }
        public virtual ICollection<Payoff> Payoffs { get; set; }
        public virtual ICollection<Transfer> Sources { get; set; }
        public virtual ICollection<Transfer> Destinations { get; set; }
    }
}

File: client-app/src/app/models/Account.ts
import dayjs from "dayjs";
import { Currency } from "./Currency";
import { AccountStatus } from "./enums/AccountStatus";

export interface Account {
    id: number;
    name: string;
    balance: number;
    convertedBalance: number;
    createdAt: Date;
    description: string;
    status: AccountStatus;
    currency: Currency;
    canBeDeleted: boolean;
}

export interface AccountFormValues {
    name: string;
    balance?: number | null;
    description: string;
    date?: dayjs.Dayjs | Date;
    currencyId?: number | string;
}